flow/Trace.cpp

```diff
@@ -1442,3 +1475,27 @@ std::string traceableStringToString(const char* value, size_t S) {
 
 	return std::string(value, S - 1); // Exclude trailing \0 byte
 }
+
+RealTimePrinter::RealTimePrinter() : beginTime(Clock::now()), flowBeginTime(g_network->now()) {}
+
+// converts the given flow time into a
+// string. now has to be larger
+// then flowBeginTime.
+// %Y-%m-%dT%H:%M:%S
+// This only has second-resolution for the simple reason
+// that std::put_time does not support higher resolution.
+// As we always also log the flow time and this is only for
+// human readability, this is good enough for now.
+std::string RealTimePrinter::toString(double now) {
+	time_t n = Clock::to_time_t(beginTime + std::chrono::microseconds(std::lround(1e6 * (now - flowBeginTime))));
```

Comment:
Is it possible to use the flow time directly since it is measured in epoch seconds? That has two benefits:

1. We won't need `g_network` upon construction, so we can support early trace events
2. We eliminate the possibility of a mismatch between the time reported in `Time` and `DateTime`.

Reply:
This makes sense to me!  I will try converting the 'now' directly